File: code/1.Two_Sum/two-sum.java
// Source : https://leetcode.com/problems/two-sum/
// Author : Kyon Huang
// Date   : 2018-02-26

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i = 0; i < nums.length; i++) {
            if(map.containsKey(nums[i]))
                return new int[]{map.get(nums[i]), i};
            map.put(target-nums[i], i);
        }
        return new int[]{0, 1};
    }
}

File: code/1.Two_Sum/two-sum.js
// Source : https://leetcode.com/problems/two-sum/
// Author : Kyon Huang
// Date   : 2017-09-16

// 第一思路，非最优解法

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    for(let i = 0; i < nums.length; i++) 
        for(let j = i + 1; j < nums.length; j++)
            if(nums[i] + nums[j] === target)
                return [i, j]
};
